framework/src/loops/ComputeJacobianBlocksThread.C

```diff
@@ -37,17 +37,46 @@ ComputeJacobianBlocksThread::~ComputeJacobianBlocksThread() {}
 void
 ComputeJacobianBlocksThread::postElement(const Elem * elem)
 {
-  std::vector<dof_id_type>
-      dof_indices; // Do this out here to avoid creating and destroying it thousands of times
+  _dof_indices.clear();
 
   Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
 
   for (const auto & block : _blocks)
   {
     const DofMap & dof_map = block->_precond_system.get_dof_map();
-    dof_map.dof_indices(elem, dof_indices);
+    dof_map.dof_indices(elem, _dof_indices);
 
     _fe_problem.addJacobianBlock(
-        block->_jacobian, block->_ivar, block->_jvar, dof_map, dof_indices, _tid);
+        block->_jacobian, block->_ivar, block->_jvar, dof_map, _dof_indices, _tid);
+  }
+}
+
+void
+ComputeJacobianBlocksThread::postInternalSide(const Elem * elem, unsigned int side)
```

Comment:
This looks like not the right fix. `ComputeJacobianThread` did the adding in `onInternalSide`, `onInterface`, `postElement`. We possibly need to move the adding in `onInternalSide`, `onInterface` out to a new function there as `postInternalSide`. Then we can override `postInternalSide` here.

Reply:
I think it is fine as is right now.

Reply:
Sounds ok to me.